pkgs/applications/editors/emacs/macport.nix

```diff
@@ -4,10 +4,10 @@
 }:
 
 stdenv.mkDerivation rec {
-  emacsVersion = "26.3";
-  emacsName = "emacs-${emacsVersion}";
+  version = "26.3";
```

Comment:
Can we change this to be even more in line with the rest of nixpkgs?

Add `pname`, `version` and remove `name` (`name` defaults to "${pname}-${version}").

Reply:
@adisbladis the challenge may be that the final package name is referencing the macport patch version as well

Reply:
Ah, right. Then make a custom `name` attribute :) Though do rename `emacsName`to `pname`.

Reply:
emacsName is used for fetching the macport patch, but I think my latest push might be good enough?

Reply:
```nix
stdenv.mkDerivation rec {
  emacsVersion = "26.3";
  macportVersion = "7.7";
  version = "${emacsVersion}-${macportVersion}";
  pname = "emacs-mac";
}
```
This should be consistent with
```
nix-repl> builtins.parseDrvName "emacs-mac-26.3-7.7"
{ name = "emacs-mac"; version = "26.3-7.7"; }
```

Reply:
This however defeats the purpose of the PR. We can't realistically expect everyone to use `emacs.emacsVersion or emacs.version`...